supporting_scripts/generate_code_cov_table/generate_code_cov_table.py

```diff
@@ -0,0 +1,224 @@
+import json
+import re
+import sys
+import argparse
+import getpass
+import os
+import logging
+
+from tqdm.auto import tqdm
+import requests
+import git
+from bs4 import BeautifulSoup
+import pyperclip
+import dotenv
+
+dotenv.load_dotenv()
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+logging.getLogger("requests").setLevel(logging.WARNING)
+logging.getLogger("urllib3").setLevel(logging.WARNING)
+
+base_url = "https://bamboo.ase.in.tum.de/"
+project_key = "ARTEMIS"
+plan_key = "TESTS"
+repo_path = "../.."  # relative to this file
+
+
+def environ_or_required(key):
+    # https://stackoverflow.com/a/45392259/4306257
+    return (
+        {'default': os.environ.get(key)} if os.environ.get(key)
+        else {'required': True}
+    )
+
+
+def get_latest_build_id(username, password, branch_name):
+    bamboo_branch_name = branch_name.replace("origin/", "").replace("/", "-")
+    url = f"{base_url}/rest/api/latest/plan/{project_key}-{plan_key}/branch/{bamboo_branch_name}.json"
+    response = requests.get(url, auth=(username, password))
+    if response.status_code != 200:
+        logging.info(f"Branch {branch_name} not found in Bamboo")
```

Comment:
This error might be a bit misleading, e.g. if Bamboo sends a response of 500 or 503. Then you would get a "branch not found" error although the issue is another. Also, I think this should be a `logging.error` here.